python/knext/knext/operator/builtin/auto_prompt.py

```diff
@@ -0,0 +1,244 @@
+            else:
+                continue
+
+        for subject_record in subject_records.values():
+            result.append(subject_record)
+        return result
+
+    def _render(self):
+        spo_infos = []
+        predicates = []
+        duplicate_types = set()
+        duplicate_predicates = set()
+        for _prop in self.property_names:
+            s_name_zh, s_desc = self.spg_type_schema_info_en.get(self.spg_type_name)
+            s_desc = (s_desc or s_name_zh) if self.spg_type_name not in duplicate_types else None
+            s_info = (s_name_zh or "") + (f'({s_desc})' if s_desc else "")
+            p_name_zh, p_desc, o_type = self.property_info_en.get(_prop)
+            p_desc = (p_desc or p_name_zh) if _prop not in duplicate_predicates else None
+            p_info = (p_name_zh or "") + (f'({p_desc})' if p_desc else "")
+            o_name_zh, o_desc = self.spg_type_schema_info_en.get(o_type)
+            o_desc = (o_desc or o_name_zh) if o_type not in duplicate_types else None
+            o_info = (o_name_zh or "") + (f'({o_desc})' if o_desc else "")
+            spo_infos.append(f'{s_info}-{p_info}-{o_info}')
+            duplicate_predicates.add(_prop)
+            duplicate_types.update([self.spg_type_name, o_type])
+            predicates.append(p_name_zh)
+        for _rel, o_type in self.relation_names:
+            s_name_zh, s_desc = self.spg_type_schema_info_en.get(self.spg_type_name)
+            s_desc = (s_desc or s_name_zh) if self.spg_type_name not in duplicate_types else None
+            s_info = (s_name_zh or "") + (f'({s_desc})' if s_desc else "")
+            p_name_zh, p_desc, _ = self.relation_info_en.get(_rel)
+            p_desc = (p_desc or p_name_zh) if _rel not in duplicate_predicates else None
+            p_info = (p_name_zh or "") + (f'({p_desc})' if p_desc else "")
+            o_name_zh, o_desc = self.spg_type_schema_info_en.get(o_type)
+            o_desc = (o_desc or o_name_zh) if o_type not in duplicate_types else None
+            o_info = (o_name_zh or "") + (f'({o_desc})' if o_desc else "")
+            spo_infos.append(f'{s_info}-{p_info}-{o_info}')
+            duplicate_predicates.add(_rel)
+            duplicate_types.update([self.spg_type_name, o_type])
+            predicates.append(p_name_zh)
+        schema_text = "\n[" + ",\n".join(spo_infos) + "]"
+        predicate_text = ','.join(predicates)
+        self.template = self.template.replace("${schema}", schema_text).replace("${predicate}", predicate_text)
+
+    def _init_render_variables(self):
+        schema_session = SchemaClient().create_session()
+        spg_type = schema_session.get(spg_type_name=self.spg_type_name)
+        self.property_info_en = {}
+        self.property_info_zh = {}
+        self.relation_info_en = {}
+        self.relation_info_zh = {}
+        self.spg_type_schema_info_en = {"Text": ("文本", None), "Integer": ("整型", None), "Float": ("浮点型", None)}
```

Comment:
we should get the Chinese and English definitions of `BasicType`s through SPG Schema API instead of hard coding.